src/activities/TakeScreenshot.ts

```diff
@@ -0,0 +1,120 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import type MapView from "@arcgis/core/views/MapView";
+import type {
+    IActivityHandler,
+    IActivityContext,
+} from "@geocortex/workflow/runtime/IActivityHandler";
+import { MapProvider } from "@geocortex/workflow/runtime/activities/arcgis/MapProvider";
+import { activate } from "@geocortex/workflow/runtime/Hooks";
+
+export interface TakeScreenshotInputs {
+    /**
+     * @description The width of the screenshot (defaults to the area width).
+     */
+    width?: number;
+
+    /**
+     * @description The height of the screenshot (defaults to the area height).
+     */
+    height: number;
+
+    /**
+     * @description The format of the resulting encoded data URL. The default is png.
+     */
+    format?: "png" | "jpg" | string;
+
+    /**
+     * @description The quality (0 to 100) of the encoded image when the format is jpg. The default is 98.
+     */
+    quality?: number;
+
+    /**
+     * @description When used, only the visible layers in this list will be included in the output.
+     */
+    layers?: __esri.Layer[];
+
+    /**
+     * @description Specifies whether to take a screenshot of a specific area of the view.
+     * The area coordinates are relative to the origin of the padded view and will be clipped to the view size. 
+     * Defaults to the whole view (padding excluded).
+     */
+    area?: {
+        /**
+         * The x coordinate of the area.
+         */
+        x?: number;
+        /**
+         * The y coordinate of the area.
+         */
+        y?: number;
+        /**
+         * The width of the area.
+         */
+        width?: number;
+        /**
+         * The height of the area.
+         */
+        height?: number;
+    };
+
+    /**
+     * @description Indicates whether to ignore the background color set in the initial view properties. The default is false.
+     */
+    ignoreBackground?: boolean;
+
+    /**
+     * @description Indicates whether the view padding should be ignored. The default is false.
+     */
+    ignorePadding?: boolean;
+}
+
+export interface TakeScreenshotOutputs {
+    /**
+     * @description A data url representing the screenshot.
+     */
+    dataUrl: string;
+
+    /**
+     * @description The raw RGBA image data.
+     */
+    data: any;
+}
+
+/**
+ * @defaultName screenshot
```

Comment:
Should be capitalized

Reply:
`@defaultName` is supposed to be camelCase. This is used to create the ID of the activity when you add it to the workflow.


`@displayName`  should be capitalized.

Reply:
And we omit the `@displayName` here on purpose because it is autogenerated from the class name and we don't need to fix any casing issues with it.

Reply:
Makes sense.